src/puya/awst/wtypes.py

```diff
@@ -545,7 +551,7 @@ def arc4_to_avm_equivalent_wtype(arc4_wtype: WType) -> WType:
         case ARC4DynamicArray(element_type=ARC4UIntN(n=8)):
             return bytes_wtype
     if arc4_wtype is arc4_string_wtype:
-        return bytes_wtype
+        return bytes_wtype  # TODO: this should be string_wtype now
```

Comment:
Whats needed to do this TODO?

Reply:
@daniel-makerx will require either changing the base type of _ABIEncoded in the stubs etc, or paring that type down as discussed, hence leaving this for another PR